components/clp-package-utils/clp_package_utils/scripts/stop_clp.py

```diff
@@ -109,47 +113,51 @@ def main(argv):
-        if "" == component_name or SEARCH_WORKER_COMPONENT_NAME == component_name:
+        if target in (CONTROLLER_TARGET_NAME, SEARCH_WORKER_COMPONENT_NAME):
             stop_container(f"clp-{SEARCH_WORKER_COMPONENT_NAME}-{instance_id}")
-        if "" == component_name or COMPRESSION_WORKER_COMPONENT_NAME == component_name:
+        if target in (CONTROLLER_TARGET_NAME, COMPRESSION_WORKER_COMPONENT_NAME):
             stop_container(f"clp-{COMPRESSION_WORKER_COMPONENT_NAME}-{instance_id}")
-        if "" == component_name or SEARCH_SCHEDULER_COMPONENT_NAME == component_name:
+        if target in (ALL_TARGET_NAME, CONTROLLER_TARGET_NAME, SEARCH_SCHEDULER_COMPONENT_NAME):
             container_name = f"clp-{SEARCH_SCHEDULER_COMPONENT_NAME}-{instance_id}"
             stop_container(container_name)
 
             container_config_file_path = logs_dir / f"{container_name}.yml"
             if container_config_file_path.exists():
                 container_config_file_path.unlink()
-        if "" == component_name or COMPRESSION_SCHEDULER_COMPONENT_NAME == component_name:
+        if target in (
+            ALL_TARGET_NAME,
+            CONTROLLER_TARGET_NAME,
+            COMPRESSION_SCHEDULER_COMPONENT_NAME,
+        ):
             container_name = f"clp-{COMPRESSION_SCHEDULER_COMPONENT_NAME}-{instance_id}"
             stop_container(container_name)
 
             container_config_file_path = logs_dir / f"{container_name}.yml"
             if container_config_file_path.exists():
                 container_config_file_path.unlink()
-        if "" == component_name or REDIS_COMPONENT_NAME == component_name:
+        if target in (ALL_TARGET_NAME, CONTROLLER_TARGET_NAME, REDIS_COMPONENT_NAME):
             container_name = f"clp-{REDIS_COMPONENT_NAME}-{instance_id}"
             stop_container(container_name)
 
             redis_config_file_path = logs_dir / f"{container_name}.conf"
             if redis_config_file_path.exists():
                 redis_config_file_path.unlink()
-        if "" == component_name or RESULTS_CACHE_COMPONENT_NAME == component_name:
+        if target in (ALL_TARGET_NAME, CONTROLLER_TARGET_NAME, RESULTS_CACHE_COMPONENT_NAME):
             container_name = f"clp-{RESULTS_CACHE_COMPONENT_NAME}-{instance_id}"
             stop_container(container_name)
-        if "" == component_name or QUEUE_COMPONENT_NAME == component_name:
+        if target in (ALL_TARGET_NAME, CONTROLLER_TARGET_NAME, QUEUE_COMPONENT_NAME):
             container_name = f"clp-{QUEUE_COMPONENT_NAME}-{instance_id}"
             stop_container(container_name)
 
             queue_config_file_path = logs_dir / f"{container_name}.conf"
             if queue_config_file_path.exists():
                 queue_config_file_path.unlink()
-        if "" == component_name or DB_COMPONENT_NAME == component_name:
+        if target in (ALL_TARGET_NAME, CONTROLLER_TARGET_NAME, DB_COMPONENT_NAME):
```

Comment:
Wait, are the database and results cache considered controller components?

Reply:
Thanks for catching it, was not intended